src/scytale/primaryHandler.go

```diff
@@ -310,3 +274,48 @@ func NewPrimaryHandler(logger log.Logger, v *viper.Viper, registry xmetrics.Regi
 
 	return router, nil
 }
+
+func doFanout(ctx context.Context, fanout *http.Request, message *wrp.Message) (context.Context, error) {
+	populateMessage(ctx, message)
+
+	bookkeepingLog(ctx, message)
```

Comment:
We don't want this here.  This is in application-layer code.  We want the bookkeeping done as part of an Alice-style decorator.  This decorator can go into the `logginghttp` package as a standard thing applications can add to their handler mappings.